php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -818,15 +828,18 @@ abstract class NDB_BVL_Instrument extends NDB_Page
         } else {
             $oldData = [];
         }
+
         $newData = array_merge($oldData ?? [], $values);
 
+        $flagDataToUpdate['Data'] = empty($newData) ? null : json_encode($newData);
```

Comment:
@CamilleBeau in order to maintain previous behaviour, you should default to and empty array, not null; **unless** you noticed an issue with defaulting to `[]`. (you are doing the exact opposite here where you are going from empty to null)

Similarly, @driusan any reason you default to `[]` and not `null` in your code before this change ?

Reply:
the coalesce to `[]` was because `array_merge` takes arrays as arguments, null is a type error.